ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@JSStats, under the tuple interpretation used in ml-proto's eval, it is neither necessary nor desirable to do that. Only a stack machine would require adjustments. But the approach you suggest would be the wrong one. Instead, you'd need to adjust whenever the type system uses subsumption (i.e., "forgets" a non-empty value in `check_type`). That occurs in more places than just branches, e.g. `(block (i32.const 0) (i32.const 1))` requires dropping the first value. Branches do not need any special treatment.